### PR TITLE
Add istiod remote to component proto

### PR DESCRIPTION
**Pull request by linsun**
related to https://github.com/istio/istio/pull/23716

### CODE REVIEW

File: operator/v1alpha1/component.proto [hunk 1]
```diff
@@ -29,7 +29,8 @@ message IstioComponentSetSpec {
     ComponentSpec policy = 33;
     ComponentSpec telemetry = 34;
     ComponentSpec cni = 38;
-
+    ComponentSpec istiodremote = 39;
```
Comment (rshriram) [inline operator/v1alpha1/component.proto:32]: istiod_remote ?

Comment (linsun) [reply operator/v1alpha1/component.proto:32]: would you prefer istiodRemote or istiod_remote?  I don't see any other name has _ in it

File: operator/v1alpha1/component.pb.go [hunk 2]
```diff
@@ -29,6 +29,7 @@ type IstioComponentSetSpec struct {
 	Policy               *ComponentSpec     `protobuf:"bytes,33,opt,name=policy,proto3" json:"policy,omitempty"`
 	Telemetry            *ComponentSpec     `protobuf:"bytes,34,opt,name=telemetry,proto3" json:"telemetry,omitempty"`
 	Cni                  *ComponentSpec     `protobuf:"bytes,38,opt,name=cni,proto3" json:"cni,omitempty"`
+	Istiodremote         *ComponentSpec     `protobuf:"bytes,39,opt,name=istiodremote,proto3" json:"istiodremote,omitempty"`
```
Comment (dcberg) [inline operator/v1alpha1/component.pb.go:32]: I'd recommend continuation of camel case (`IstiodRemote`).

File: operator/v1alpha1/component.pb.go [hunk 3]
```diff
@@ -96,6 +97,13 @@ func (m *IstioComponentSetSpec) GetCni() *ComponentSpec {
 	return nil
 }
 
+func (m *IstioComponentSetSpec) GetIstiodremote() *ComponentSpec {
```
Comment (dcberg) [inline operator/v1alpha1/component.pb.go:100]: Same camel case comment .... `GetIstiodRemote`.

File: python/istio_api/operator/v1alpha1/component_pb2.py [hunk 4]
```diff
@@ -779,15 +786,16 @@
...
+  serialized_start=3171,
+  serialized_end=3191,
 )
 
 _ISTIOCOMPONENTSETSPEC.fields_by_name['base'].message_type = _BASECOMPONENTSPEC
 _ISTIOCOMPONENTSETSPEC.fields_by_name['pilot'].message_type = _COMPONENTSPEC
 _ISTIOCOMPONENTSETSPEC.fields_by_name['policy'].message_type = _COMPONENTSPEC
 _ISTIOCOMPONENTSETSPEC.fields_by_name['telemetry'].message_type = _COMPONENTSPEC
 _ISTIOCOMPONENTSETSPEC.fields_by_name['cni'].message_type = _COMPONENTSPEC
+_ISTIOCOMPONENTSETSPEC.fields_by_name['istiodremote'].message_type = _COMPONENTSPEC
```
Comment (dcberg) [inline python/istio_api/operator/v1alpha1/component_pb2.py:798]: To remain consistent with the pattern consider `istiod_remote`.

File: operator/v1alpha1/component.proto (hunk 1 above)
Comment (linsun) [reply operator/v1alpha1/component.proto]: updated with thanks!

File: operator/v1alpha1/component.pb.go (hunk 2 above)
Comment (linsun) [reply operator/v1alpha1/component.pb.go]: should be updated as well - this file is generated.

File: operator/v1alpha1/component.pb.go (hunk 3 above)
Comment (linsun) [reply operator/v1alpha1/component.pb.go]: should be updated as well - this file is generated.

File: python/istio_api/operator/v1alpha1/component_pb2.py (hunk 4 above)
Comment (linsun) [reply python/istio_api/operator/v1alpha1/component_pb2.py]: should be updated as well - this file is generated.

**Comment by linsun** [comment]
@dcberg I have updated and regenerated all files.  Thanks!

**Pull request merged**